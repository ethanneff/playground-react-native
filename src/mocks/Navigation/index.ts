import { jest } from '@jest/globals';
export const mockNavigate = jest.fn();
export const mockGoBack = jest.fn();
