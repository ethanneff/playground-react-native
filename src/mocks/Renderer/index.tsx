import {
  BaseNavigationContainer,
  NavigationContext,
  NavigationRouteContext,
} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React, {ReactNode} from 'react';
import {Provider} from 'react-redux';
import {create, ReactTestRenderer} from 'react-test-renderer';
import {store} from '../../redux/core';

type MockNavigation = any;
type MockRenderer = {
  tree: ReactTestRenderer;
  navigation: MockNavigation;
};
type Props = {
  component: ReactNode;
  route?: {key: string; name: string};
  navigation?: MockNavigation;
};

export const mockRenderer = ({
  component,
  route = {key: '', name: ''},
  navigation = {
    goBack: jest.fn(),
    navigate: jest.fn(),
    push: jest.fn(),
    popToTop: jest.fn(),
    addListener: () => jest.fn(),
    isFocused: jest.fn().mockReturnValue(true),
    reset: jest.fn(),
    emit: jest.fn(),
  },
}: Props): MockRenderer => {
  const Stack = createStackNavigator();

  const Component = () => (
    <NavigationContext.Provider value={navigation}>
      <NavigationRouteContext.Provider value={route}>
        {component}
      </NavigationRouteContext.Provider>
    </NavigationContext.Provider>
  );

  const Providers = () => (
    <Provider store={store}>
      <BaseNavigationContainer>
        <Stack.Navigator>
          <Stack.Screen component={Component} name="MockedScreen" />
        </Stack.Navigator>
      </BaseNavigationContainer>
    </Provider>
  );

  const tree = create(<Providers />);

  return {tree, navigation};
};
