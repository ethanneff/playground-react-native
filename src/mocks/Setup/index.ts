import { NativeModules } from 'react-native';
import 'react-native-gesture-handler/jestSetup';

jest.mock('@invertase/react-native-apple-authentication', () => () => ({
  appleAuth: jest.fn(),
}));

const mockUserInfo = {
  idToken: 'mockIdToken',
  accessToken: null,
  accessTokenExpirationDate: null,
  serverAuthCode: 'mockServerAuthCode',
  scopes: [],
  user: {
    email: 'mockEmail',
    id: 'mockId',
    givenName: 'mockGivenName',
    familyName: 'mockFamilyName',
    photo: 'mockPhotoUtl',
    name: 'mockFullName',
  },
};

jest.mock('@react-native-community/google-signin', () => ({
  GoogleSignin: {
    configure: jest.fn(),
    hasPlayServices: jest.fn(() => Promise.resolve(true)),
    signIn: jest.fn(() => Promise.resolve(mockUserInfo)),
    signInSilently: jest.fn(() => Promise.resolve(mockUserInfo)),
    revokeAccess: jest.fn(() => Promise.resolve(true)),
    signOut: jest.fn(() => Promise.resolve(true)),
  },
}));

jest.mock('@react-native-firebase/analytics', () => () => ({
  logEvent: jest.fn(),
  setUserProperties: jest.fn(),
  setUserId: jest.fn(),
  setCurrentScreen: jest.fn(),
}));

jest.mock('react-native-sound', () => {
  return class Mock {
    constructor() {}
    setVolume = jest.fn();
    setNumberOfLoops = jest.fn();
    play = jest.fn();
    stop = jest.fn();
    static setCategory = jest.fn();
  };
});

jest.mock('react-native-device-info', () => ({
  getAPILevel: jest.fn(),
  getApplicationName: jest.fn(),
  getBatteryLevel: jest.fn(() => Promise.resolve(1)),
  getBrand: jest.fn(),
  getBuildNumber: jest.fn(),
  getBundleId: jest.fn(),
  getCarrier: jest.fn(),
  getDeviceCountry: jest.fn(),
  getDeviceId: jest.fn(),
  getDeviceLocale: jest.fn(),
  getDeviceName: jest.fn(),
  getFirstInstallTime: jest.fn(),
  getFontScale: jest.fn(),
  getFreeDiskStorage: jest.fn(),
  getInstallReferrer: jest.fn(),
  getInstanceID: jest.fn(),
  getLastUpdateTime: jest.fn(),
  getManufacturer: jest.fn(),
  getMaxMemory: jest.fn(),
  getModel: jest.fn(),
  getPhoneNumber: jest.fn(),
  getReadableVersion: jest.fn(),
  getSerialNumber: jest.fn(),
  getSystemName: jest.fn(),
  getSystemVersion: jest.fn(),
  getTimezone: jest.fn(() => 'America/Los_Angeles'),
  getTotalDiskCapacity: jest.fn(),
  getTotalMemory: jest.fn(),
  getUniqueID: jest.fn(),
  getUserAgent: jest.fn(),
  getVersion: jest.fn(),
  is24Hour: jest.fn(),
  isEmulator: jest.fn(),
  isPinOrFingerprintSet: jest.fn(),
  isTablet: jest.fn(() => false),
}));

NativeModules.RNCNetInfo = {
  addListener: jest.fn(),
  getCurrentState: jest.fn(() => Promise.resolve()),
  removeListeners: jest.fn(),
};

NativeModules.RNCAsyncStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeChecklistItem: jest.fn(),
  mergeItem: jest.fn(),
  clear: jest.fn(),
  getAllKeys: jest.fn(),
  flushGetRequests: jest.fn(),
  multiGet: jest.fn(),
  multiSet: jest.fn(),
  multiRemove: jest.fn(),
  multiMerge: jest.fn(),
};

jest.useFakeTimers();

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};
  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
