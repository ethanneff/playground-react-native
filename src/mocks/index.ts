export * from './Navigation';
export * from './Renderer';
