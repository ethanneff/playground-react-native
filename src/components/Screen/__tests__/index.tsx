import React from 'react';
import { View } from 'react-native';
import { Screen } from '..';
import { mockRenderer } from '../../../mocks/Renderer';

describe('screen', () => {
  it('renders correctly', () => {
    expect.hasAssertions();
    const { tree } = mockRenderer({
      component: (
        <Screen>
          <View />
        </Screen>
      ),
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
