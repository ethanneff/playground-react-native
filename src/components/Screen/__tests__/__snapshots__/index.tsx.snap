// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      undefined,
      Object {
        "backgroundColor": "hsl(0,0%,100%)",
        "flex": 1,
      },
    ]
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "left",
        "right",
        "bottom",
      ]
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0,0%,100%)",
            "borderBottomColor": "hsl(0,0%,100%)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 8,
            "zIndex": 2,
          },
        ]
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={1}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "flex": 0,
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "justifyContent": "center",
                  },
                  null,
                ],
              ]
            }
            testID="leftNav"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "transparent",
                    "fontSize": 32,
                  },
                  Array [
                    undefined,
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰅁
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={2}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "flex": 0,
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "justifyContent": "center",
                  },
                  Object {
                    "alignSelf": "flex-end",
                  },
                ],
              ]
            }
            testID="rightNav"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "transparent",
                    "fontSize": 32,
                  },
                  Array [
                    undefined,
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰅖
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "hsl(0,0%,100%)",
            "height": 2,
            "width": "100%",
            "zIndex": 1,
          },
        ]
      }
    />
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </RNCSafeAreaView>
</View>
`;
